irohad/synchronization/synchronization.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//
+// Created by Takumi Yamashita on 2017/04/28.
+//
+
+#ifndef IROHA_SYNCHRONIZATION_SYNCHRONIZATION_H
+#define IROHA_SYNCHRONIZATION_SYNCHRONIZATION_H
+
+#include <functional>
+#include <network/sync_client.hpp>
+#include <network/sync_gate.hpp>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include <model/model.hpp>
+
+namespace iroha {
+  namespace synchronization {
+
+    using Block = iroha::model::Block;
+
+    class BlockCompare : public std::binary_function<Block, Block, bool> {
```

Comment:
It's probably better to have a Block comparator as a util in some other place, as there is a general need for this.